File: python/design-patterns/adapter/adapter/adapter.py
class LegacyBankAdapter:   # implements our PaymentGateway shape
    def __init__(self, bank: LegacyBank):
        self.bank = bank

    def pay(self, amount_usd: float):
        cents = round(amount_usd * 100)
        res = self.bank.make_payment(cents, "USD")
        return {"ok": res["status"] == 0, "ref": res["txn"]}

File: cpp/design-patterns/adapter/adapter/adapter.cpp
class LegacyBankAdapter : public PaymentGateway {
    LegacyBank bank;
public:
    Receipt pay(double amountUsd) override {
        int cents = (int)(amountUsd * 100 + 0.5);
        auto [status, txn] = bank.makePayment(cents, "USD");
        return { status == 0, txn };
    }
};